src/steamship/experimental/package_starters/telegram_bot.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Optional, Type
+
+import requests
+from pydantic import Field
+
+from steamship import SteamshipError
+from steamship.invocable import Config, InvocableResponse, PackageService, post
+
+
+class TelegramBotConfig(Config):
+    bot_token: str = Field(description="The secret token for your Telegram bot")
+
+
+class TelegramBot(PackageService, ABC):
+
+    config: TelegramBotConfig
+
+    @classmethod
+    def config_cls(cls) -> Type[Config]:
+        """Return the Configuration class."""
+        return TelegramBotConfig
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.api_root = f"https://api.telegram.org/bot{self.config.bot_token}"
+
+    def instance_init(self):
+        """This instance init method is called automatically when an instance of this package is created. It registers the URL of the instance as the Telegram webhook for messages."""
+        webhook_url = self.context.invocable_url + "respond"
+        response = requests.get(
+            f"{self.api_root}/setWebhook",
+            params={"url": webhook_url, "allowed_updates": ["message"]},
+        )
+        if not response.ok:
+            raise SteamshipError(
+                f"Could not set webhook for bot. Webhook URL was {webhook_url}. Telegram response message: {response.text}"
+            )
+        logging.info(f"Initialized webhook with URL {webhook_url}")
+
+    @abstractmethod
+    def respond_to_text(self, message_text: str, chat_id: int, message_id: int) -> Optional[str]:
```

Comment:
i think this is OK for now, but I'm wondering why not pass through the other args as **kwargs here too (in case there is something like group info or user info needed)?

def respond_to_text(self, message_text: str, chat_id: int, message_id: int, **kwargs) -> Optional[str]:

and

response = self.respond_to_text(message_text, chat_id, message_id, **kwargs)

It allows a backdoor abuse channel for the contract, but provides a decent amount of flexibility.

Reply:
Sure!